portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java

```diff
@@ -3506,6 +3527,20 @@ protected Folder moveFolder(
 	@BeanReference(type = RepositoryProvider.class)
 	protected RepositoryProvider repositoryProvider;
 
+	private void _copyResourcePermissions(
+			long companyId, String className, long sourceResourcePrimKey,
+			long targetResourcePrimKey)
+		throws PortalException {
+
+		for (int scope : ResourceConstants.SCOPES) {
+			_resourcePermissionLocalService.deleteResourcePermissions(
+				companyId, className, scope, targetResourcePrimKey);
+		}
+
+		_resourceLocalService.copyModelResources(
+			companyId, className, sourceResourcePrimKey, targetResourcePrimKey);
```

Comment:
Can you directly use the method `_resourcePermissionLocalService.copyModelResourcePermissions` and get rid of `_resourceLocalService` ?

Reply:
Done.